modules/k8s_eks_addons/ingress-nginx.tf

```diff
@@ -18,9 +18,12 @@ resource "helm_release" "ingress_nginx" {
   dependency_update = true
   values = [
     templatefile("${path.module}/templates/nginx_values.yaml", {
-      public_subnets = join(", ", var.ingress_nginx_config.subnets_ids)
+      public_subnets         = join(", ", var.ingress_nginx_config.subnets_ids)
+      protocol               = var.aws_load_balancer_controller_config.enable ? "ssl" : "tcp"
+      aws_load_balancer_type = var.aws_load_balancer_controller_config.enable ? "external" : "nlb"
     }),
-    var.ingress_nginx_config.chart_values
+    var.aws_load_balancer_controller_config.enable ? var.aws_load_balancer_controller_config.chart_values : var.ingress_nginx_config.chart_values
   ]
-  timeout = 1200
+  timeout    = 1200
+  depends_on = [helm_release.aws_load_balancer_controller, kubernetes_namespace_v1.ingress_nginx]
```

Comment:
I would assume that there is an implicit dependency since the namespace is referenced in line 12.

Reply:
Thank you for pointing that out. I have removed it